lib/Factory/LayoutFactory.php

```diff
@@ -288,6 +289,65 @@ public function getById($layoutId)
         return $layouts[0];
     }
 
+    /**
+     * Get CampaignId from layout history
+     * @param int $layoutId
+     * @return int campaignId|null
+     * @throws NotFoundException
+     */
+    public function getCampaignId($layoutId)
```

Comment:
Maybe we should call this `getCampaignIdFromLayoutHistory()` just so it is clear where it goes for that information?